zokrates_core/src/static_analysis/propagation.rs

```diff
@@ -302,10 +327,124 @@ impl<'ast, T: Field> Folder<'ast, T> for Propagator<'ast, T> {
                     c => ArrayExpressionInner::IfElse(box c, box consequence, box alternative),
                 }
             }
+            ArrayExpressionInner::Member(box s, m) => {
+                let s = self.fold_struct_expression(s);
+
+                let members = match s.get_type() {
+                    Type::Struct(members) => members,
+                    _ => unreachable!(),
+                };
+
+                match s.into_inner() {
+                    StructExpressionInner::Value(v) => {
+                        match members.iter().zip(v).find(|(id, _)| id.0 == m).unwrap().1 {
+                            TypedExpression::Array(a) => a.into_inner(),
+                            _ => unreachable!(),
+                        }
+                    }
+                    inner => ArrayExpressionInner::Member(box inner.annotate(members), m),
+                }
+            }
             e => fold_array_expression_inner(self, ty, size, e),
         }
     }
 
+    fn fold_struct_expression_inner(
+        &mut self,
+        ty: &Vec<(MemberId, Type)>,
+        e: StructExpressionInner<'ast, T>,
+    ) -> StructExpressionInner<'ast, T> {
+        match e {
+            StructExpressionInner::Identifier(id) => {
+                match self
+                    .constants
+                    .get(&TypedAssignee::Identifier(Variable::struc(
```

Comment:
```suggestion
                    .get(&TypedAssignee::Identifier(Variable::struct(
```

Reply:
`struct` is reserved in Rust, cannot be used for a function name